x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart_annotations.ts

```diff
@@ -132,6 +132,19 @@ export function renderAnnotations(
   const levelHeight = ANNOTATION_LEVEL_HEIGHT;
   const levels = getAnnotationLevels(focusAnnotationData);
 
+  const onAnnotationMouseOver = function (this: object, d: Annotation) {
+    showFocusChartTooltip(d, this);
+  };
+
+  const onAnnotationClick = (d: Annotation) => {
+    // clear a possible existing annotation set up for editing before setting the new one.
+    // this needs to be done explicitly here because a new annotation created using the brush tool
+    // could still be present in the chart.
+    annotationUpdatesService.setValue(null);
+    // set the actual annotation and trigger the flyout
+    annotationUpdatesService.setValue(d);
```

Comment:
I see you added a comment with an explanation, but it seems odd to me anyway. 🤔 Have you considered putting this logic in `setValue` method instead? I will definitely make an API of `annotationUpdatesService ` more straightforward.

Reply:
Looking through the code base it does seems like we use `annotationUpdatesService.setValue(null)`. Consolidating the logic inside `setValue` sounds like a good idea but I think it's probably better to do in a follow up as it's not neccessarily a bug fix.